rfd/0219-k8s-rbac-simplification.md

```diff
@@ -0,0 +1,242 @@
+
+#### Error management
+
+The error messages when using `kubectl` will be improved to include a link to
+the documentation and more details on what is expected. This will help with
+initial custom setups that skipped the provided provisioning scripts.
+
+#### New preset role
+
+To help onboarding, as currently no preset role grants Kubernetes access,
+a new preset role will be added, `kube-editor`, which will have wildcard access
+to Kubernetes resources as well as the predefined `kubernetes_groups` mapping
+to the cluster admin group.
+
+```yaml
+kind: role
+version: v8
+metadata:
+  name: kube-editor
+spec:
+  allow:
+    kubernetes_groups:
+      - teleport-cluster-admin
+    kubernetes_labels:
+      '*': '*'
+    kubernetes_resources:
+      - api_group: '*'
+        kind: '*'
+        namespace: '*'
+        name: '*'
+        verbs:
+          - '*'
+```
+
+### User flow
+
+#### Current flow
+
+- Day 0 - Initial setup:
+  - User installs Teleport and sets up a Kubernetes cluster.
+  - User provisions the Kubernetes cluster using a script, discovery, or Helm
+    chart to create the impersonation ClusterRole/ClusterRoleBinding.
+  - User configures Kubernetes RBAC with a custom ClusterRole with some
+    permissions
+  - User configures Kubernetes RBAC with a custom ClusterRoleBinding, which
+    needs to be understood matches with the `kubernetes_groups` field in the
+    Teleport role.
+  - User configures Teleport role with `kubernetes_groups`, (unclear what
+    `kubernetes_users` does from the docs) fields to match the custom
+    ClusterRoleBinding.
+  - User configures Teleport role with `kubernetes_resources` and
+    `kubernetes_labels` to match to match or reduce the permissions granted by
+    the custom ClusterRole.
+- Day 1 - Ongoing management:
+  - User needs to understand the interaction between the Teleport role and the
+    underlying Kubernetes RBAC.
+  - User may need to update the Teleport role if the underlying Kubernetes
+    RBAC changes.
+  - No clear guidance on how to which permission should be set where (Teleport
+    or Kubernetes).
+  - User may need to troubleshoot unexpected permissions or access issues.
+
+#### Proposed flow
+
+- Day 0 - Initial setup:
+  - User installs Teleport and sets up a Kubernetes cluster.
+  - User provisions the Kubernetes cluster using a script, discovery, or Helm
+    chart to create all required RBAC resources.
+  - User configures the Teleport role with `kubernetes_resources` and
+    `kubernetes_labels` to match or reduce the permissions granted by the
+    ClusterRole. The `kubernetes_groups` field is pre-populated in the Web UI
+    and well documented in the Teleport documentation for the YAML version.
+- Day 1 - Ongoing management:
+  - User can manage the Teleport role without needing to understand the
```

Comment:
if that's the case, what is the purpose of `kubernetes_groups` in the preset role `kube-editor`?

Reply:
While it could be removed for new customers, it's purpose it to maintain compatibility with existing customers who may not be willing to apply the new clusterrole/clusterrolebinding to enable the simplification.